bluepyefe/formats/spike2.py

```diff
@@ -0,0 +1,444 @@
+"""Axon reader"""
+
+"""
+Copyright (c) 2020, EPFL/Blue Brain Project
+
+ This file is part of BluePyEfe <https://github.com/BlueBrain/BluePyEfe>
+
+ This library is free software; you can redistribute it and/or modify it under
+ the terms of the GNU Lesser General Public License version 3.0 as published
+ by the Free Software Foundation.
+
+ This library is distributed in the hope that it will be useful, but WITHOUT
+ ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+ details.
+
+ You should have received a copy of the GNU Lesser General Public License
+ along with this library; if not, write to the Free Software Foundation, Inc.,
+ 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+"""
+
+from neo import io
+from collections import OrderedDict
+import logging
+logger = logging.getLogger(__name__)
+import numpy as np
+import os
+import json
+import sys
+import quantities
+from quantities import ms, s, mV, nA
+import matplotlib.pyplot as plt
+import matplotlib.gridspec as gridspec
+from neo import AnalogSignal
+from neo.io import Spike2IO
+from . import common
+
+
+def process(config=None,
+            filename=None,
+            cellname=None,
+            expname=None,
+            stim_feats=None,
+            idx_file=None,
+            ljp=0, v_corr=0):
+
+    path = config['path']
+    cells = config['cells']
+
+    # initialize data dictionary
+    data = common.manageDicts.initialize_data_dict()
+
+    # read metadata file if present
+    meta_dict = {}
+
+    f_meta = os.path.join(path, cellname, filename + '_metadata.json')
+
+    #
+    if os.path.exists(f_meta):
+        with open(f_meta) as json_metadata:
+            meta_dict = json.load(json_metadata)
+        json_metadata.close()
+    else:
+        logger.debug("No metadata file found for file %s. \
+                Skipping file", filename)
+        print("No metadata found. Skipping file " + filename)
```

Comment:
Hello @lbologna, thank you for the pull request.

Could you uniformize the use of logger and print in this file ? 
It would be best if all the outputs were handled using the logger.

Thank you

Reply:
Hello @DrTaDa,

Yes sure, that went unnoticed. I will also do a global check on this and the common.py file.
Thanks.

Reply:
Hello again @lbologna,

I am sorry to be finicky about that. In the cases where the logging message is after an exception, the level of the logger should be error and not debug. Otherwise, the error message will not appear if the user did not set it's logger to DEBUG.

Reply:
Hello @DrTaDa,

no problem. Thanks for spotting this.
Just changed the logger level to "error" for exceptions.